gohcl/decode_test.go

```diff
@@ -535,7 +741,7 @@ func TestDecodeBody(t *testing.T) {
 				t.Fatalf("diagnostics while parsing: %s", diags.Error())
 			}
 
-			targetVal := reflect.New(reflect.TypeOf(test.Target))
+			targetVal := test.Target().(reflect.Value)
```

Comment:
If we make the various other functions return just normal values, we can get the `reflect.Value` we need here like this:

```go
targetVal := reflect.ValueOf(test.Target())
```